xdsl/xdsl_opt_main.py

```diff
@@ -222,19 +223,20 @@ def parse_mlir(io: IO[str]):
         self.available_frontends['xdsl'] = parse_xdsl
         self.available_frontends['mlir'] = parse_mlir
 
+    def register_pass(self, opPass: Type[OperationPass]):
+        self.available_passes[opPass.name] = opPass
+
     def register_all_passes(self):
         """
         Register all passes that can be used.
 
         Add other/additional passes by overloading this function.
         """
-        self.available_passes['lower-mpi'] = lower_mpi
-        self.available_passes[
-            'convert-stencil-to-ll-mlir'] = ConvertStencilToLLMLIR
-        self.available_passes['convert-stencil-to-gpu'] = ConvertStencilToGPU
-        self.available_passes[
-            'stencil-shape-inference'] = StencilShapeInference
-        self.available_passes['frontend-desymrefy'] = Desymrefy
+        self.register_pass(LowerMPIPass)
```

Comment:
I personally preferred to old way. This gave you the information of how to call a given class in more straight-forward, central manner, IMO.

Reply:
I get that. Note that it is displayed in `xdsl-opt --help`

Reply:
Exactly, I looked into printing more complete help there too, just gave up on output formatting details because all that is already a sidetrack for me.
But then, we could have something like `xdsl-opt --help passes` displaying extensive information from docstring or whatever we think makes sense

Reply:
@webmiche, are you okay with this change or are you strongly for keeping the names here?

Reply:
I am fine with it, as long as we add the necessary --help support for it in the near future.

Reply:
Note that xdsl-opt --help already prints all available passes' names.
It would be lovely to print more information somewhere there, such as the docstring if present? I will see if I can do something on that side.